test/kyberGovernance.js

```diff
@@ -0,0 +1,727 @@
+      Helper.assertEqualArray(eventLogs.args.withDelegatecalls, withDelegatecalls);
+
+      let options = ['YES', 'NO'];
+      let voteCounts = [0, 0];
+      // check correct proposal data
+      await Helper.assertEqual(proposalCount.add(new BN(1)), await governance.getProposalsCount());
+      let proposalData = await governance.getProposalById(proposalCount);
+      Helper.assertEqual(proposalData.id, proposalCount);
+      Helper.assertEqual(proposalData.proposalType, 1); // binary is 0
+      Helper.assertEqual(proposalData.creator, daoOperator);
+      Helper.assertEqual(proposalData.executor, executor.address);
+      Helper.assertEqual(proposalData.strategy, votingStrategy.address);
+      Helper.assertEqualArray(proposalData.targets, targets);
+      Helper.assertEqualArray(proposalData.weiValues, weiValues);
+      Helper.assertEqualArray(proposalData.signatures, signatures);
+      Helper.assertEqualArray(proposalData.calldatas, calldatas);
+      Helper.assertEqualArray(proposalData.withDelegatecalls, withDelegatecalls);
+      Helper.assertEqualArray(proposalData.options, options);
+      Helper.assertEqualArray(proposalData.voteCounts, voteCounts);
+      Helper.assertEqual(proposalData.totalVotes, 0);
+      Helper.assertEqual(proposalData.maxVotingPower, maxVotingPower);
+      Helper.assertEqual(proposalData.startTime, startTime);
+      Helper.assertEqual(proposalData.endTime, endTime);
+      Helper.assertEqual(proposalData.executionTime, 0);
+      Helper.assertEqual(proposalData.link, link);
+      Helper.assertEqual(proposalData.executed, false);
+      Helper.assertEqual(proposalData.canceled, false);
+    });
+  });
+
+  const createGenericProposal = async (executor, strategy, options, startTime, endTime, link) => {
+    let proposalCount = await governance.getProposalsCount();
+    await governance.createGenericProposal(executor, strategy, options, startTime, endTime, link);
+    return proposalCount;
+  };
+
+  const createBinaryProposal = async (
+    executor,
+    strategy,
+    targets,
+    weiValues,
+    signatures,
+    calldatas,
+    withDelegatecalls,
+    startTime,
+    endTime,
+    link
+  ) => {
+    let proposalCount = await governance.getProposalsCount();
+    await governance.createBinaryProposal(
+      executor,
+      strategy,
+      [targets, weiValues, signatures, calldatas, withDelegatecalls],
+      startTime,
+      endTime,
+      link
+    );
+    return proposalCount;
+  };
+
+  describe('#test cancel proposals', async () => {
+    beforeEach('setup', async () => {
+      governance = await KyberGovernance.new(
+        admin,
+        daoOperator,
+        [validator.address, executor.address],
+        [votingStrategy.address]
+      );
+    });
+
+    it('reverts invalid proposal id', async () => {
+      let proposalCount = await governance.getProposalsCount();
+      await expectRevert.unspecified(governance.cancel(proposalCount.add(new BN(1))));
```

Comment:
This should be `invalid proposal id`, right?

Reply:
yeah we can add check for proposal id in range